beacon_node/src/config.rs

```diff
@@ -809,11 +761,6 @@ pub fn get_config<E: EthSpec>(
         }
     }
 
-    // Note: This overrides any previous flags that enable this option.
-    if cli_args.get_flag("disable-deposit-contract-sync") {
-        client_config.sync_eth1_chain = false;
-    }
```

Comment:
We should also deprecate this rather than fully removing it (log a warning here)